skfda/exploratory/visualization/representation.py

```diff
@@ -79,249 +132,402 @@ def _get_color_info(fdata, group, group_names, group_colors, legend, kwargs):
-            `group` will be used.
-        **kwargs: if dim_domain is 1, keyword arguments to be passed to
-            the matplotlib.pyplot.plot function; if dim_domain is 2,
-            keyword arguments to be passed to the
-            matplotlib.pyplot.plot_surface function.
-
-    Returns:
-        fig (figure object): figure object in which the graphs are plotted.
+        fdata: functional data set that we want to plot.
+        gradient_color_list: list of real values used to determine the color
+            in which each of the instances will be plotted.
+        max_grad: maximum value that the gradient_list can take, it will be
+            used to normalize the ``gradient_color_list``. If not
+            declared it will be initialized to the maximum value of
+            gradient_list
+        min_grad: minimum value that the gradient_list can take, it will be
+            used to normalize the ``gradient_color_list``. If not
+            declared it will be initialized to the minimum value of
+            gradient_list.
 
     """
 
-    fig, axes = _get_figure_and_axes(chart, fig, axes)
-    fig, axes = _set_figure_layout_for_fdata(fdata, fig, axes, n_rows, n_cols)
+    def __init__(
+        self,
+        fdata: FData,
+        gradient_color_list: Optional[Sequence[float]] = None,
+        max_grad: Optional[float] = None,
+        min_grad: Optional[float] = None,
+    ) -> None:
+        self.fdata = fdata
+        self.gradient_color_list = gradient_color_list
+        if self.gradient_color_list is not None:
+            if len(self.gradient_color_list) != fdata.n_samples:
+                raise ValueError(
+                    "The length of the gradient color"
+                    "list should be the same as the number"
+                    "of samples in fdata",
+                )
+
+            if min_grad is None:
+                self.min_grad = min(self.gradient_color_list)
+            else:
+                self.min_grad = min_grad
+
+            if max_grad is None:
+                self.max_grad = max(self.gradient_color_list)
+            else:
+                self.max_grad = max_grad
+
+            aux_list = [
+                grad_color - self.min_grad
+                for grad_color in self.gradient_color_list
+            ]
+
+            self.gradient_list: Sequence[float] = (
+                [
+                    aux / (self.max_grad - self.min_grad)
+                    for aux in aux_list
+                ]
+            )
+        else:
+            self.gradient_list = []
+
+    def plot(
+        self,
+        chart: Union[Figure, Axes, None] = None,
+        *,
+        fig: Optional[Figure] = None,
+        ax: Optional[Axes] = None,
+        n_rows: Optional[int] = None,
+        n_cols: Optional[int] = None,
+        n_points: Union[int, Tuple[int, int], None] = None,
+        domain_range: Optional[DomainRangeLike] = None,
+        group: Union[Sequence[Any], None] = None,
```

Comment:
I do not get why these parameters are in `plot` instead of init. Also, how they interact with the gradients?

Reply:
the thing was that i was going to change that in the next pull, in the branch outliergram_interactive its changed in some